In [1]:
import numpy as pd
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
app_train = pd.read_csv('../data/application_train.csv')
app_test = pd.read_csv('../data/application_test.csv')

In [3]:
app_train

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0           100002       1         Cash loans           M            N   
1           100003       0         Cash loans           F            N   
2           100004       0    Revolving loans           M            Y   
3           100006       0         Cash loans           F            N   
4           100007       0         Cash loans           M            N   
...            ...     ...                ...         ...          ...   
307506      456251       0         Cash loans           M            N   
307507      456252       0         Cash loans           F            N   
307508      456253       0         Cash loans           F            N   
307509      456254       1         Cash loans           F            N   
307510      456255       0         Cash loans           F            N   

       FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  \
0                    Y             0          202500.0    406597.5   
1                    N             0          270000.0   1293502.5   
2                    Y             0           67500.0    135000.0   
3                    Y             0          135000.0    312682.5   
4                    Y             0          121500.0    513000.0   
...                ...           ...               ...         ...   
307506               N             0          157500.0    254700.0   
307507               Y             0           72000.0    269550.0   
307508               Y             0          153000.0    677664.0   
307509               Y             0          171000.0    370107.0   
307510               N             0          157500.0    675000.0   

        AMT_ANNUITY  ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
0           24700.5  ...                 0                0                0   
1           35698.5  ...                 0                0                0   
2            6750.0  ...                 0                0                0   
3           29686.5  ...                 0                0                0   
4           21865.5  ...                 0                0                0   
...             ...  ...               ...              ...              ...   
307506      27558.0  ...                 0                0                0   
307507      12001.5  ...                 0                0                0   
307508      29979.0  ...                 0                0                0   
307509      20205.0  ...                 0                0                0   
307510      49117.5  ...                 0                0                0   

       FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                     0                        0.0                       0.0   
1                     0                        0.0                       0.0   
2                     0                        0.0                       0.0   
3                     0                        NaN                       NaN   
4                     0                        0.0                       0.0   
...                 ...                        ...                       ...   
307506                0                        NaN                       NaN   
307507                0                        NaN                       NaN   
307508                0                        1.0                       0.0   
307509                0                        0.0                       0.0   
307510                0                        0.0                       0.0   

        AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                              0.0                        0.0   
1                              0.0                        0.0   
2                              0.0                        0.0   
3                              NaN                        NaN   
4                              

In [4]:
app_test

SK_ID_CURR NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  \
0          100001         Cash loans           F            N               Y   
1          100005         Cash loans           M            N               Y   
2          100013         Cash loans           M            Y               Y   
3          100028         Cash loans           F            N               Y   
4          100038         Cash loans           M            Y               N   
...           ...                ...         ...          ...             ...   
48739      456221         Cash loans           F            N               Y   
48740      456222         Cash loans           F            N               N   
48741      456223         Cash loans           F            Y               Y   
48742      456224         Cash loans           M            N               N   
48743      456250         Cash loans           F            Y               N   

       CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                 0          135000.0    568800.0      20560.5   
1                 0           99000.0    222768.0      17370.0   
2                 0          202500.0    663264.0      69777.0   
3                 2          315000.0   1575000.0      49018.5   
4                 1          180000.0    625500.0      32067.0   
...             ...               ...         ...          ...   
48739             0          121500.0    412560.0      17473.5   
48740             2          157500.0    622413.0      31909.5   
48741             1          202500.0    315000.0      33205.5   
48742             0          225000.0    450000.0      25128.0   
48743             0          135000.0    312768.0      24709.5   

       AMT_GOODS_PRICE  ... FLAG_DOCUMENT_18 FLAG_DOCUMENT_19  \
0             450000.0  ...                0                0   
1             180000.0  ...                0                0   
2             630000.0  ...                0                0   
3            1575000.0  ...                0                0   
4             625500.0  ...                0                0   
...                ...  ...              ...              ...   
48739         270000.0  ...                0                0   
48740         495000.0  ...                0                0   
48741         315000.0  ...                0                0   
48742         450000.0  ...                0                0   
48743         270000.0  ...                0                0   

      FLAG_DOCUMENT_20 FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR  \
0                    0                0                        0.0   
1                    0                0                        0.0   
2                    0                0                        0.0   
3                    0                0                        0.0   
4                    0                0                        NaN   
...                ...              ...                        ...   
48739                0                0                        0.0   
48740                0                0                        NaN   
48741                0                0                        0.0   
48742                0                0                        0.0   
48743                0                0                        0.0   

       AMT_REQ_CREDIT_BUREAU_DAY  AMT_REQ_CREDIT_BUREAU_WEEK  \
0                            0.0                         0.0   
1                            0.0                         0.0   
2                            0.0                         0.0   
3                            0.0                         0.0   
4                            NaN                         NaN   
...                          ...                         ...   
48739                        0.0                         0.0   
48740                        NaN                         NaN   
48741                        0.0           

In [5]:
bureau_balance_file = pd.read_csv('../data/bureau_balance.csv')

In [6]:
bureau_balance_file

SK_ID_BUREAU  MONTHS_BALANCE STATUS
0              5715448               0      C
1              5715448              -1      C
2              5715448              -2      C
3              5715448              -3      C
4              5715448              -4      C
...                ...             ...    ...
27299920       5041336             -47      X
27299921       5041336             -48      X
27299922       5041336             -49      X
27299923       5041336             -50      X
27299924       5041336             -51      X

[27299925 rows x 3 columns]

In [7]:
bureau_file = pd.read_csv('../data/bureau.csv')

In [8]:
bureau_file

SK_ID_CURR  SK_ID_BUREAU CREDIT_ACTIVE CREDIT_CURRENCY  DAYS_CREDIT  \
0            215354       5714462        Closed      currency 1         -497   
1            215354       5714463        Active      currency 1         -208   
2            215354       5714464        Active      currency 1         -203   
3            215354       5714465        Active      currency 1         -203   
4            215354       5714466        Active      currency 1         -629   
...             ...           ...           ...             ...          ...   
1716423      259355       5057750        Active      currency 1          -44   
1716424      100044       5057754        Closed      currency 1        -2648   
1716425      100044       5057762        Closed      currency 1        -1809   
1716426      246829       5057770        Closed      currency 1        -1878   
1716427      246829       5057778        Closed      currency 1         -463   

         CREDIT_DAY_OVERDUE  DAYS_CREDIT_ENDDATE  DAYS_ENDDATE_FACT  \
0                         0               -153.0             -153.0   
1                         0               1075.0                NaN   
2                         0                528.0                NaN   
3                         0                  NaN                NaN   
4                         0               1197.0                NaN   
...                     ...                  ...                ...   
1716423                   0                -30.0                NaN   
1716424                   0              -2433.0            -2493.0   
1716425                   0              -1628.0             -970.0   
1716426                   0              -1513.0            -1513.0   
1716427                   0                  NaN             -387.0   

         AMT_CREDIT_MAX_OVERDUE  CNT_CREDIT_PROLONG  AMT_CREDIT_SUM  \
0                           NaN                   0        91323.00   
1                           NaN                   0       225000.00   
2                           NaN                   0       464323.50   
3                           NaN                   0        90000.00   
4                       77674.5                   0      2700000.00   
...                         ...                 ...             ...   
1716423                     0.0                   0        11250.00   
1716424                  5476.5                   0        38130.84   
1716425                     NaN                   0        15570.00   
1716426                     NaN                   0        36000.00   
1716427                     NaN                   0        22500.00   

         AMT_CREDIT_SUM_DEBT  AMT_CREDIT_SUM_LIMIT  AMT_CREDIT_SUM_OVERDUE  \
0                        0.0                   NaN                     0.0   
1                   171342.0                   NaN                     0.0   
2                        NaN                   NaN                     0.0   
3                        NaN                   NaN                     0.0   
4                        NaN                   NaN                     0.0   
...                      ...                   ...                     ...   
1716423              11250.0                   0.0                     0.0   
1716424                  0.0                   0.0                     0.0   
1716425                  NaN                   NaN                     0.0   
1716426                  0.0                   0.0                     0.0   
1716427                  0.0                   NaN                     0.0   

             CREDIT_TYPE  DAYS_CREDIT_UPDATE  AMT_ANNUITY  
0        Consumer credit                -131          NaN  
1            Credit card                 -20          NaN  
2        Consumer credit                 -16          NaN  
3            Credit card                 -16          NaN  
4        Consumer credit                 -21          NaN  
...                  ...                 ... 

In [9]:
credit_card_balanbce_file = pd.read_csv('../data/credit_card_balance.csv')
credit_card_balanbce_file

SK_ID_PREV  SK_ID_CURR  MONTHS_BALANCE  AMT_BALANCE  \
0           2562384      378907              -6       56.970   
1           2582071      363914              -1    63975.555   
2           1740877      371185              -7    31815.225   
3           1389973      337855              -4   236572.110   
4           1891521      126868              -1   453919.455   
...             ...         ...             ...          ...   
3840307     1036507      328243              -9        0.000   
3840308     1714892      347207              -9        0.000   
3840309     1302323      215757              -9   275784.975   
3840310     1624872      430337             -10        0.000   
3840311     2411345      236760             -10        0.000   

         AMT_CREDIT_LIMIT_ACTUAL  AMT_DRAWINGS_ATM_CURRENT  \
0                         135000                       0.0   
1                          45000                    2250.0   
2                         450000                       0.0   
3                         225000                    2250.0   
4                         450000                       0.0   
...                          ...                       ...   
3840307                    45000                       NaN   
3840308                    45000                       0.0   
3840309                   585000                  270000.0   
3840310                   450000                       NaN   
3840311                   157500                       0.0   

         AMT_DRAWINGS_CURRENT  AMT_DRAWINGS_OTHER_CURRENT  \
0                       877.5                         0.0   
1                      2250.0                         0.0   
2                         0.0                         0.0   
3                      2250.0                         0.0   
4                     11547.0                         0.0   
...                       ...                         ...   
3840307                   0.0                         NaN   
3840308                   0.0                         0.0   
3840309              270000.0                         0.0   
3840310                   0.0                         NaN   
3840311                   0.0                         0.0   

         AMT_DRAWINGS_POS_CURRENT  AMT_INST_MIN_REGULARITY  ...  \
0                           877.5                 1700.325  ...   
1                             0.0                 2250.000  ...   
2                             0.0                 2250.000  ...   
3                             0.0                11795.760  ...   
4                         11547.0                22924.890  ...   
...                           ...                      ...  ...   
3840307                       NaN                    0.000  ...   
3840308                       0.0                    0.000  ...   
3840309                       0.0                 2250.000  ...   
3840310                       NaN                    0.000  ...   
3840311                       0.0                    0.000  ...   

         AMT_RECIVABLE  AMT_TOTAL_RECEIVABLE  CNT_DRAWINGS_ATM_CURRENT  \
0                0.000                 0.000                       0.0   
1            64875.555             64875.555                       1.0   
2            31460.085             31460.085                       0.0   
3           233048.970            233048.970                       1.0   
4           453919.455            453919.455                       0.0   
...                ...                   ...                       ...   
3840307          0.000                 0.000                       NaN   
3840308          0.000                 0.000                       0.0   
3840309     273093.975            273093.975                       2.0   
3840310          0.000                 0.000                       NaN   
3840311          0.000                 0.000                       0.0   

         CNT_DRAWINGS_CURRENT  CNT_DRAWINGS_OTHER_CURRENT  \
0      

In [10]:
HomeCredit_file = pd.read_csv('../data/HomeCredit_columns_description.csv', encoding='latin')
HomeCredit_file

Unnamed: 0                         Table                    Row  \
0             1  application_{train|test}.csv             SK_ID_CURR   
1             2  application_{train|test}.csv                 TARGET   
2             5  application_{train|test}.csv     NAME_CONTRACT_TYPE   
3             6  application_{train|test}.csv            CODE_GENDER   
4             7  application_{train|test}.csv           FLAG_OWN_CAR   
..          ...                           ...                    ...   
214         217     installments_payments.csv  NUM_INSTALMENT_NUMBER   
215         218     installments_payments.csv        DAYS_INSTALMENT   
216         219     installments_payments.csv     DAYS_ENTRY_PAYMENT   
217         220     installments_payments.csv         AMT_INSTALMENT   
218         221     installments_payments.csv            AMT_PAYMENT   

                                           Description  \
0                             ID of loan in our sample   
1    Target variable (1 - client with payment diffi...   
2          Identification if loan is cash or revolving   
3                                 Gender of the client   
4                        Flag if the client owns a car   
..                                                 ...   
214            On which installment we observe payment   
215  When the installment of previous credit was su...   
216  When was the installments of previous credit p...   
217  What was the prescribed installment amount of ...   
218  What the client actually paid on previous cred...   

                                   Special  
0                                      NaN  
1                                      NaN  
2                                      NaN  
3                                      NaN  
4                                      NaN  
..                                     ...  
214                                    NaN  
215  time only relative to the application  
216  time only relative to the application  
217                                    NaN  
218                                    NaN  

[219 rows x 5 columns]

In [11]:
installments_payments_file = pd.read_csv('../data/installments_payments.csv', encoding='latin')
installments_payments_file

SK_ID_PREV  SK_ID_CURR  NUM_INSTALMENT_VERSION  \
0            1054186      161674                     1.0   
1            1330831      151639                     0.0   
2            2085231      193053                     2.0   
3            2452527      199697                     1.0   
4            2714724      167756                     1.0   
...              ...         ...                     ...   
13605396     2186857      428057                     0.0   
13605397     1310347      414406                     0.0   
13605398     1308766      402199                     0.0   
13605399     1062206      409297                     0.0   
13605400     2448869      434321                     1.0   

          NUM_INSTALMENT_NUMBER  DAYS_INSTALMENT  DAYS_ENTRY_PAYMENT  \
0                             6          -1180.0             -1187.0   
1                            34          -2156.0             -2156.0   
2                             1            -63.0               -63.0   
3                             3          -2418.0             -2426.0   
4                             2          -1383.0             -1366.0   
...                         ...              ...                 ...   
13605396                     66          -1624.0                 NaN   
13605397                     47          -1539.0                 NaN   
13605398                     43             -7.0                 NaN   
13605399                     43          -1986.0                 NaN   
13605400                     19            -27.0                 NaN   

          AMT_INSTALMENT  AMT_PAYMENT  
0               6948.360     6948.360  
1               1716.525     1716.525  
2              25425.000    25425.000  
3              24350.130    24350.130  
4               2165.040     2160.585  
...                  ...          ...  
13605396          67.500          NaN  
13605397          67.500          NaN  
13605398       43737.435          NaN  
13605399          67.500          NaN  
13605400       11504.250          NaN  

[13605401 rows x 8 columns]

In [12]:
POS_CASH_balance_file = pd.read_csv('../data/POS_CASH_balance.csv', encoding='latin')
POS_CASH_balance_file

SK_ID_PREV  SK_ID_CURR  MONTHS_BALANCE  CNT_INSTALMENT  \
0            1803195      182943             -31            48.0   
1            1715348      367990             -33            36.0   
2            1784872      397406             -32            12.0   
3            1903291      269225             -35            48.0   
4            2341044      334279             -35            36.0   
...              ...         ...             ...             ...   
10001353     2448283      226558             -20             6.0   
10001354     1717234      141565             -19            12.0   
10001355     1283126      315695             -21            10.0   
10001356     1082516      450255             -22            12.0   
10001357     1259607      174278             -52            16.0   

          CNT_INSTALMENT_FUTURE NAME_CONTRACT_STATUS  SK_DPD  SK_DPD_DEF  
0                          45.0               Active       0           0  
1                          35.0               Active       0           0  
2                           9.0               Active       0           0  
3                          42.0               Active       0           0  
4                          35.0               Active       0           0  
...                         ...                  ...     ...         ...  
10001353                    0.0               Active     843           0  
10001354                    0.0               Active     602           0  
10001355                    0.0               Active     609           0  
10001356                    0.0               Active     614           0  
10001357                    0.0            Completed       0           0  

[10001358 rows x 8 columns]

In [15]:
previous_application_file = pd.read_csv('../data/previous_application.csv', encoding='latin')
previous_application_file

SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  \
0           2030495      271877     Consumer loans     1730.430   
1           2802425      108129         Cash loans    25188.615   
2           2523466      122040         Cash loans    15060.735   
3           2819243      176158         Cash loans    47041.335   
4           1784265      202054         Cash loans    31924.395   
...             ...         ...                ...          ...   
1670209     2300464      352015     Consumer loans    14704.290   
1670210     2357031      334635     Consumer loans     6622.020   
1670211     2659632      249544     Consumer loans    11520.855   
1670212     2785582      400317         Cash loans    18821.520   
1670213     2418762      261212         Cash loans    16431.300   

         AMT_APPLICATION  AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE  \
0                17145.0     17145.0               0.0          17145.0   
1               607500.0    679671.0               NaN         607500.0   
2               112500.0    136444.5               NaN         112500.0   
3               450000.0    470790.0               NaN         450000.0   
4               337500.0    404055.0               NaN         337500.0   
...                  ...         ...               ...              ...   
1670209         267295.5    311400.0               0.0         267295.5   
1670210          87750.0     64291.5           29250.0          87750.0   
1670211         105237.0    102523.5           10525.5         105237.0   
1670212         180000.0    191880.0               NaN         180000.0   
1670213         360000.0    360000.0               NaN         360000.0   

        WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  ...  \
0                         SATURDAY                       15  ...   
1                         THURSDAY                       11  ...   
2                          TUESDAY                       11  ...   
3                           MONDAY                        7  ...   
4                         THURSDAY                        9  ...   
...                            ...                      ...  ...   
1670209                  WEDNESDAY                       12  ...   
1670210                    TUESDAY                       15  ...   
1670211                     MONDAY                       12  ...   
1670212                  WEDNESDAY                        9  ...   
1670213                     SUNDAY                       10  ...   

         NAME_SELLER_INDUSTRY  CNT_PAYMENT  NAME_YIELD_GROUP  \
0                Connectivity         12.0            middle   
1                         XNA         36.0        low_action   
2                         XNA         12.0              high   
3                         XNA         12.0            middle   
4                         XNA         24.0              high   
...                       ...          ...               ...   
1670209             Furniture         30.0        low_normal   
1670210             Furniture         12.0            middle   
1670211  Consumer electronics         10.0        low_normal   
1670212                   XNA         12.0        low_normal   
1670213                   XNA         48.0            middle   

                 PRODUCT_COMBINATION  DAYS_FIRST_DRAWING DAYS_FIRST_DUE  \
0           POS mobile with interest            365243.0          -42.0   
1                   Cash X-Sell: low            365243.0         -134.0   
2                  Cash X-Sell: high            365243.0         -271.0   
3                Cash X-Sell: middle            365243.0         -482.0   
4                  Cash Street: high                 NaN            NaN   
...                              ...                 ...            ...   
1670209   POS industry with interest            365243.0         -508.0   
1670210   POS industry with interest            365243.0        -1604.0   
1670211  POS household with interest      

In [16]:
sample_submission_file = pd.read_csv('../data/sample_submission.csv', encoding='latin')
sample_submission_file

SK_ID_CURR  TARGET
0          100001     0.5
1          100005     0.5
2          100013     0.5
3          100028     0.5
4          100038     0.5
...           ...     ...
48739      456221     0.5
48740      456222     0.5
48741      456223     0.5
48742      456224     0.5
48743      456250     0.5

[48744 rows x 2 columns]

In [17]:
cols_intersect_app_train_vs_bureau_balance = app_train.columns.intersection(bureau_balance_file.columns)

In [18]:
cols_intersect_app_train_vs_bureau_balance

Index([], dtype='object')

In [19]:
col_intersect_app_train_vs_bureau = app_train.columns.intersection(bureau_file.columns)
col_intersect_app_train_vs_bureau

Index(['SK_ID_CURR', 'AMT_ANNUITY'], dtype='object')

In [20]:
HomeCredit_file[['Row','Description']].head(50)

Row  \
0                    SK_ID_CURR   
1                        TARGET   
2            NAME_CONTRACT_TYPE   
3                   CODE_GENDER   
4                  FLAG_OWN_CAR   
5               FLAG_OWN_REALTY   
6                  CNT_CHILDREN   
7              AMT_INCOME_TOTAL   
8                    AMT_CREDIT   
9                   AMT_ANNUITY   
10              AMT_GOODS_PRICE   
11              NAME_TYPE_SUITE   
12             NAME_INCOME_TYPE   
13          NAME_EDUCATION_TYPE   
14           NAME_FAMILY_STATUS   
15            NAME_HOUSING_TYPE   
16   REGION_POPULATION_RELATIVE   
17                   DAYS_BIRTH   
18                DAYS_EMPLOYED   
19            DAYS_REGISTRATION   
20              DAYS_ID_PUBLISH   
21                  OWN_CAR_AGE   
22                   FLAG_MOBIL   
23               FLAG_EMP_PHONE   
24              FLAG_WORK_PHONE   
25             FLAG_CONT_MOBILE   
26                   FLAG_PHONE   
27                   FLAG_EMAIL   
28              OCCUPATION_TYPE   
29              CNT_FAM_MEMBERS   
30         REGION_RATING_CLIENT   
31  REGION_RATING_CLIENT_W_CITY   
32   WEEKDAY_APPR_PROCESS_START   
33      HOUR_APPR_PROCESS_START   
34   REG_REGION_NOT_LIVE_REGION   
35   REG_REGION_NOT_WORK_REGION   
36  LIVE_REGION_NOT_WORK_REGION   
37       REG_CITY_NOT_LIVE_CITY   
38       REG_CITY_NOT_WORK_CITY   
39      LIVE_CITY_NOT_WORK_CITY   
40            ORGANIZATION_TYPE   
41                 EXT_SOURCE_1   
42                 EXT_SOURCE_2   
43                 EXT_SOURCE_3   
44               APARTMENTS_AVG   
45             BASEMENTAREA_AVG   
46  YEARS_BEGINEXPLUATATION_AVG   
47              YEARS_BUILD_AVG   
48               COMMONAREA_AVG   
49                ELEVATORS_AVG   

                                          Description  
0                            ID of loan in our sample  
1   Target variable (1 - client with payment diffi...  
2         Identification if loan is cash or revolving  
3                                Gender of the client  
4                       Flag if the client owns a car  
5                 Flag if client owns a house or flat  
6                   Number of children the client has  
7                                Income of the client  
8                           Credit amount of the loan  
9                                        Loan annuity  
10  For consumer loans it is the price of the good...  
11  Who was accompanying client when he was applyi...  
12  Clients income type (businessman, working, mat...  
13     Level of highest education the client achieved  
14                        Family status of the client  
15  What is the housing situation of the client (r...  
16  Normalized population of region where client l...  
17    Client's age in days at the time of application  
18  How many days before the application the perso...  
19  How many days before the application did clien...  
20  How many days before the application did clien...  
21                                Age of client's car  
22      Did client provide mobile phone (1=YES, 0=NO)  
23        Did client provide work phone (1=YES, 0=NO)  
24        Did client provide home phone (1=YES, 0=NO)  
25           Was mobile phone reachable (1=YES, 0=NO)  
26        Did client provide home phone (1=YES, 0=NO)  
27             Did client provide email (1=YES, 0=NO)  
28       What kind of occupation does the client have  
29           How many family members does client have  
30  Our rating of the region where client lives (1...  
31  Our rating of the region where client lives wi...  
32  On which day of the week did the client apply ...  
33  Approximately at what hour did the client appl...  
34  Flag if client's permanent address does not ma...  
35  Flag if client's permanent address does not ma...  
36  Flag if client's contact address does not matc...  
37  Flag if client's permanent address does not ma...  
38  Flag if client's permanent address does not ma...  
39

In [21]:
HomeCredit_file.tail(50)

Unnamed: 0                      Table                          Row  \
169         172    credit_card_balance.csv    CNT_INSTALMENT_MATURE_CUM   
170         173    credit_card_balance.csv         NAME_CONTRACT_STATUS   
171         174    credit_card_balance.csv                       SK_DPD   
172         175    credit_card_balance.csv                   SK_DPD_DEF   
173         176   previous_application.csv                  SK_ID_PREV    
174         177   previous_application.csv                   SK_ID_CURR   
175         178   previous_application.csv           NAME_CONTRACT_TYPE   
176         179   previous_application.csv                  AMT_ANNUITY   
177         180   previous_application.csv              AMT_APPLICATION   
178         181   previous_application.csv                   AMT_CREDIT   
179         182   previous_application.csv             AMT_DOWN_PAYMENT   
180         183   previous_application.csv              AMT_GOODS_PRICE   
181         184   previous_application.csv   WEEKDAY_APPR_PROCESS_START   
182         185   previous_application.csv      HOUR_APPR_PROCESS_START   
183         186   previous_application.csv  FLAG_LAST_APPL_PER_CONTRACT   
184         187   previous_application.csv       NFLAG_LAST_APPL_IN_DAY   
185         188   previous_application.csv             NFLAG_MICRO_CASH   
186         189   previous_application.csv            RATE_DOWN_PAYMENT   
187         190   previous_application.csv        RATE_INTEREST_PRIMARY   
188         191   previous_application.csv     RATE_INTEREST_PRIVILEGED   
189         192   previous_application.csv       NAME_CASH_LOAN_PURPOSE   
190         193   previous_application.csv         NAME_CONTRACT_STATUS   
191         194   previous_application.csv                DAYS_DECISION   
192         195   previous_application.csv            NAME_PAYMENT_TYPE   
193         196   previous_application.csv           CODE_REJECT_REASON   
194         197   previous_application.csv              NAME_TYPE_SUITE   
195         198   previous_application.csv             NAME_CLIENT_TYPE   
196         199   previous_application.csv          NAME_GOODS_CATEGORY   
197         200   previous_application.csv               NAME_PORTFOLIO   
198         201   previous_application.csv            NAME_PRODUCT_TYPE   
199         202   previous_application.csv                 CHANNEL_TYPE   
200         203   previous_application.csv             SELLERPLACE_AREA   
201         204   previous_application.csv         NAME_SELLER_INDUSTRY   
202         205   previous_application.csv                  CNT_PAYMENT   
203         206   previous_application.csv             NAME_YIELD_GROUP   
204         207   previous_application.csv          PRODUCT_COMBINATION   
205         208   previous_application.csv           DAYS_FIRST_DRAWING   
206         209   previous_application.csv               DAYS_FIRST_DUE   
207         210   previous_application.csv    DAYS_LAST_DUE_1ST_VERSION   
208         211   previous_application.csv                DAYS_LAST_DUE   
209         212   previous_application.csv             DAYS_TERMINATION   
210         213   previous_application.csv    NFLAG_INSURED_ON_APPROVAL   
211         214  installments_payments.csv                  SK_ID_PREV    
212         215  installments_payments.csv                   SK_ID_CURR   
213         216  installments_payments.csv       NUM_INSTALMENT_VERSION   
214         217  installments_payments.csv        NUM_INSTALMENT_NUMBER   
215         218  installments_payments.csv              DAYS_INSTALMENT   
216         219  installments_payments.csv           DAYS_ENTRY_PAYMENT   
217         220  installments_payments.csv               AMT_INSTALMENT   
218         221  installments_payments.csv                  AMT_PAYMENT   

                                           Description  \
169  Number of paid installments on the previous cr...   
170  Contract status (active signed,...) on the pre...   
171  

In [22]:
app_train.head()

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   
2      100004       0    Revolving loans           M            Y   
3      100006       0         Cash loans           F            N   
4      100007       0         Cash loans           M            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   
2               Y             0           67500.0    135000.0       6750.0   
3               Y             0          135000.0    312682.5      29686.5   
4               Y             0          121500.0    513000.0      21865.5   

   ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20 FLAG_DOCUMENT_21  \
0  ...                 0                0                0                0   
1  ...                 0                0                0                0   
2  ...                 0                0                0                0   
3  ...                 0                0                0                0   
4  ...                 0                0                0                0   

  AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                        0.0                       0.0   
1                        0.0                       0.0   
2                        0.0                       0.0   
3                        NaN                       NaN   
4                        0.0                       0.0   

   AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                         0.0                        0.0   
1                         0.0                        0.0   
2                         0.0                        0.0   
3                         NaN                        NaN   
4                         0.0                        0.0   

   AMT_REQ_CREDIT_BUREAU_QRT  AMT_REQ_CREDIT_BUREAU_YEAR  
0                        0.0                         1.0  
1                        0.0                         0.0  
2                        0.0                         0.0  
3                        NaN                         NaN  
4                        0.0                         0.0  

[5 rows x 122 columns]

In [23]:
app_train.tail()

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
307506      456251       0         Cash loans           M            N   
307507      456252       0         Cash loans           F            N   
307508      456253       0         Cash loans           F            N   
307509      456254       1         Cash loans           F            N   
307510      456255       0         Cash loans           F            N   

       FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  \
307506               N             0          157500.0    254700.0   
307507               Y             0           72000.0    269550.0   
307508               Y             0          153000.0    677664.0   
307509               Y             0          171000.0    370107.0   
307510               N             0          157500.0    675000.0   

        AMT_ANNUITY  ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
307506      27558.0  ...                 0                0                0   
307507      12001.5  ...                 0                0                0   
307508      29979.0  ...                 0                0                0   
307509      20205.0  ...                 0                0                0   
307510      49117.5  ...                 0                0                0   

       FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
307506                0                        NaN                       NaN   
307507                0                        NaN                       NaN   
307508                0                        1.0                       0.0   
307509                0                        0.0                       0.0   
307510                0                        0.0                       0.0   

        AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
307506                         NaN                        NaN   
307507                         NaN                        NaN   
307508                         0.0                        1.0   
307509                         0.0                        0.0   
307510                         0.0                        2.0   

        AMT_REQ_CREDIT_BUREAU_QRT  AMT_REQ_CREDIT_BUREAU_YEAR  
307506                        NaN                         NaN  
307507                        NaN                         NaN  
307508                        0.0                         1.0  
307509                        0.0                         0.0  
307510                        0.0                         1.0  

[5 rows x 122 columns]

Nous avons 307510 prets caractererisé par 122 critères decrit dans le fichier Home_credit

In [24]:
app_train['TARGET'].unique()

array([1, 0], dtype=int64)

TARGET est la variable cible avec :
1 :  le client n'a pas pu remboursé son pret (Non solvable)
0 :  le client a pu remboursé son pret (solvable)

In [25]:
app_train['NAME_CONTRACT_TYPE'].unique()

array(['Cash loans', 'Revolving loans'], dtype=object)

on a deux type de contrat de pret :
Cash loans : si le pret est cash d'un coup
Revolving : si le pret est par tranche

In [26]:
app_train[['NAME_CONTRACT_TYPE','TARGET']].value_counts(normalize=True)*100

NAME_CONTRACT_TYPE  TARGET
Cash loans          0         82.927440
Revolving loans     0          8.999678
Cash loans          1          7.551275
Revolving loans     1          0.521607
Name: proportion, dtype: float64

In [27]:
counts = app_train['NAME_CONTRACT_TYPE'].isna().sum()
print(counts)

0


on a poresque 90% de demande de pret cash et presk 10% et dans 90% de cash preté 83 % n'ont pas pu remboursé et dans les 10% les 9 % n'ont pas pu rembousé.

In [28]:
app_train.CODE_GENDER.unique()

array(['M', 'F', 'XNA'], dtype=object)

In [29]:
app_train['CODE_GENDER'].value_counts()

CODE_GENDER
F      202448
M      105059
XNA         4
Name: count, dtype: int64

M: Masculin
F : Feminin
XNA : NON identifié

In [30]:
app_train.FLAG_OWN_CAR.unique()

array(['N', 'Y'], dtype=object)

Y :le demandeur est proproetaire de sa voiture
N : le demandeur n'est pas propriétaire 

In [31]:
app_train.FLAG_OWN_REALTY.unique()

array(['Y', 'N'], dtype=object)

Y : le client a une maison
N : le client n'a pas de maison

In [32]:
app_train.AMT_GOODS_PRICE.unique()

array([ 351000. , 1129500. ,  135000. , ...,  453465. ,  143977.5,
        743863.5])